### DISCUSSION TITLE
Retrieving mdast information from a docusaurus plugin

### DISCUSSION THREAD

**Discussion by chanjunren** [facebook/docusaurus#10430] [Q&A]
Hi, I'm currently looking into implementing a graph view (docusaurus) [Obsidian's graph view](https://help.obsidian.md/Plugins/Graph+view)

I have written a remark plugin that converts some of obsidian's markdown syntax to docusaurus's markdown syntax, and I was wondering if I would be able to access the processed mdast from a docusaurus plugin

The overall approach I have in mind is as follows:
1. Traverse the processed mdast to create 'nodes' for each markdown document (a node can be a link to another document or a markdown tag)
2. setting some globalData for what nodes a markdown document is linked to
3. create a custom Component to render this graph

I have tried looking into Docusaurus's plugin documentation to see if accessing the generated mdast is possible, but have not found anything concrete.

Any advice is greatly appreciated 🙏🏻

**Comment by slorber** [comment]
Not sure if we have all the low-level apis to make this possible.

Afaik some users (@anaclumos ?) built something like this in userland as a pre-build step.

All our 3 content plugins support Remark plugins, and Remark plugins have access to the mdast, running inside our mdx loader during the bundling phase. An idea could be to collect your link graph in remark plugins, and then expose it as a JSON file created in a `postBuild` hook? Haven't tried but this could work.

**Reply by anaclumos** [reply]
I'm flattered to be mentioned!

I used to have a graph view but then removed it couple weeks ago due to its poor performance. It was rather a hacky implementation of statically reading and parsing links and feeding them into a graph library like D3.

I run this script before git push: https://github.com/anaclumos/extracranial/blob/main/tools/process-backlinks.py

Which updates this file: https://github.com/anaclumos/extracranial/blob/main/src/data/backlinks.ts

While I removed this atm, I'd like to revisit this topic tho.

**Reply by chanjunren** [reply]
@slorber @anaclumos Thank you for your replies! 
After further looking into the Docusaurus lifecycle methods, I have decided to change my approach and I have made some progress by pre-processing my markdown files in my Docusaurus plugin, and outputting a JSON that can be read and used by my remark plugin

@anaclumos I looked through your repository and it seemed to me that you use obsidian to write your documents as well and docusaurus to publish them (which is the same as me!)

I'm currently working on a mini project as I use obsidian to write my markdown notes and wanted to have some of obsidian's features on my docusaurus website. If you are keen maybe we could exchange ideas and collaborate?

**Reply by anaclumos** [reply]
SGTM!

**Reply by fsferrara** [reply]
@chanjunren I am doing the same, following @anaclumos example 😄 
I was wondering if it worth to create a sort of vault template for us to share.

**Reply by chanjunren** [reply]
Hi @anaclumos @fsferrara , I've just published an MVP of my [my plugin](https://github.com/chanjunren/vaultusaurus), feel free to check it out and let me know what you guys think!

You can see it in action [here](https://chanjunren.github.io/docs/zettelkasten/snowflake_id)

**Reply by slorber** [reply]
looks cool @chanjunren thanks for sharing. Let me know if anything in Docusaurus could simplify your integration.
